index.bs

```diff
@@ -3159,21 +3168,32 @@ Each {{WebRequestEvent}} has:
           1. Set |details|'s {{WebRequestBeforeRequestDetails/requestBody}}
               to |requestBody|.
 
-      1. If |handler|'s [=WebRequest handler config/blocking=] flag is true,
-          then:
+      1. [=Queue a global task=] on |taskSource| that will run the
+          following steps:
 
           1. Let |result| be the result of calling |handler|[[=WebRequest
               handler config/handler=]] given |details|.
 
-          1. If any [=map/key=] in |result| is not [=list/contained=] in the set
-              « "cancel", "redirect" », then [=throw=] a {{TypeError}}.
+              Note: Any exceptions thrown by the handler are intentionally
```

Comment:
is there a spec version of try / catch?

Reply:
Not that I know of. If we want to catch a JS exception we can check the return value of the script execution call, which we don't need to do here.